### PR TITLE
Changed line color of boxplot for dark_background

### LINKED ISSUE #15675: Boxplot line color with style dark_background should be bright
### Bug report

```
%pylab inline
plt.style.use('dark_background')
plt.boxplot([1,2,3])
```

The line color should not be black. With the dark_background style the boxes are not visible.

### PR DESCRIPTION
**Pull request by manan2501**
## PR Summary

Rework of #15705. Closes #15675. 
Previously, the black outlines of boxes in boxplots were hidden inside the black background in the `dark_background` style.

Now: Outputs of boxplot_demo.py (https://matplotlib.org/examples/pylab_examples/boxplot_demo.html) using dark_background are as shown below:
image
image
image
image
image
image
image

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by ImportanceOfBeingErnest** [comment]
I think the style should only set those properties that are necessary to change for the dark background. E.g. the linewidth does not care if it's a bright or dark background. This would allow to combine different styles.

**Comment by manan2501** [comment]
> I think the style should only set those properties that are necessary to change for the dark background. E.g. the linewidth does not care if it's a bright or dark background. This would allow to combine different styles.

I have copied the whole thing from classic.mplstyle and just edited colors.
I haven't touched the linewidth.

**Comment by manan2501** [comment]
> I think the style should only set those properties that are necessary to change for the dark background. E.g. the linewidth does not care if it's a bright or dark background. This would allow to combine different styles.

Do you mean to say linestyle?

**Comment by ImportanceOfBeingErnest** [comment]
That's precisely the point. The dark_background style should not contain linewidth, nor any other parameter that is not necessary for dark backgrounds.

**Comment by ImportanceOfBeingErnest** [comment]
Any reason to make the mean and median green instead of the original color from the colorcyle?

**Comment by manan2501** [comment]
> Any reason to make the mean and median green instead of the original color from the colorcyle?

I just thought that it would look highlighted. No specific reason. 
If it's unappropriate than I will change it?

File: lib/matplotlib/mpl-data/stylelib/dark_background.mplstyle [hunk 1]
```diff
@@ -21,3 +21,13 @@ figure.edgecolor: black
 savefig.facecolor: black
 savefig.edgecolor: black
 
+### Boxplots
+boxplot.boxprops.color: w
```
Comment (timhoffm) [inline lib/matplotlib/mpl-data/stylelib/dark_background.mplstyle]: Please use `white`. While technically `w` is allowed, we try to keep style files more readable.

File: lib/matplotlib/mpl-data/stylelib/dark_background.mplstyle [hunk 2]
```diff
@@ -21,3 +21,10 @@ figure.edgecolor: black
 savefig.facecolor: black
 savefig.edgecolor: black
 
+### Boxplots
+boxplot.boxprops.color: white
+boxplot.capprops.color: white
+boxplot.flierprops.color: white
+boxplot.flierprops.markeredgecolor: white
+boxplot.meanprops.markeredgecolor: white
```
Comment (timhoffm) [inline lib/matplotlib/mpl-data/stylelib/dark_background.mplstyle]: `boxplot.meanprops.markerfacecolor` and `boxplot.meanprops.markeredgecolor` are both `C2` 
Therefore, I think we do not need to redefine `boxplot.meanprops.markeredgecolor`.

Comment (tacaswell) [inline lib/matplotlib/mpl-data/stylelib/dark_background.mplstyle:29]:
```suggestion
```

Per @timhoffm 's suggestion.  I think it is better to track the color cycle for these so they stay highlighted from the other lines.

**Comment by timhoffm** [comment]
I took the liberty of pushing a commit for not setting `boxplot.flierprops.markeredgecolor` as discussed above.

**Comment by QuLogic** [comment]
Going to ignore doc failures, since they're fixed on master.

**Pull request merged**